src/STEMMUS_SCOPE.m

```diff
@@ -339,12 +339,26 @@
         % update GroundwaterSettings.headBotmLayer and GroundwaterSettings.tempBotm, from MODFLOW through BMI
         GroundwaterSettings.gw_Dep = groundwater.calculateGroundWaterDepth(GroundwaterSettings.topLevel, GroundwaterSettings.headBotmLayer, ModelSettings.Tot_Depth);
 
-        % update Dunnian runoff and ForcingData.Precip_msr
+        % Update Dunnian runoff and ForcingData.Precip_msr
         [ForcingData.R_Dunn, ForcingData.Precip_msr] = groundwater.updateDunnianRunoff(ForcingData.Precip_msr, GroundwaterSettings.gw_Dep);
 
         % Calculate the index of the bottom layer level
         [GroundwaterSettings.indxBotmLayer, GroundwaterSettings.indxBotmLayer_R] = groundwater.calculateIndexBottomLayer(GroundwaterSettings.soilThick, GroundwaterSettings.gw_Dep);
         [depToGWT_strt, indxGWLay_strt] = groundwater.findPhreaticSurface(SoilVariables.hh, KT, GroundwaterSettings.soilThick, GroundwaterSettings.indxBotmLayer_R);
+
+        % Check the position of the groundwater table
+        if any(GroundwaterSettings.gw_Dep > ModelSettings.Tot_Depth) || any(isempty(GroundwaterSettings.indxBotmLayer))
+            warning('Groundwater table is below the end of the soil column. Please enlarge the total soil thickness!');
+            return
+        end
+
+        % Update soil temperature
+        if ~isnan(GroundwaterSettings.tempBotm)
+            SoilVariables.T(1:GroundwaterSettings.indxBotmLayer) = GroundwaterSettings.tempBotm;
+            T = SoilVariables.T;
+            SoilVariables.TT(1:GroundwaterSettings.indxBotmLayer) = GroundwaterSettings.tempBotm;
+            TT = SoilVariables.TT;
+        end
```

Comment:
```suggestion
        % no available groundwater temperature data
        if isnan(GroundwaterSettings.tempBotm)
            GroundwaterSettings.tempBotm = SoilVariables.TT(GroundwaterSettings.indxBotmLayer);
        end
```

Reply:
the other if statement does not affect the results because `RHS` is set in [this line](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/09887a13d938c6478babe853c4c88ed4ebccff0a/src/%2Benergy/solveTridiagonalMatrixEquations.m#L27-L31). 
@MostafaGomaa93 can you please remove [this line](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/09887a13d938c6478babe853c4c88ed4ebccff0a/src/%2Benergy/calculateBoundaryConditions.m#L21-L23)? I moved it here.

Reply:
> the other if statement does not affect the results because `RHS` is set in [this line](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/09887a13d938c6478babe853c4c88ed4ebccff0a/src/%2Benergy/solveTridiagonalMatrixEquations.m#L27-L31). @MostafaGomaa93 can you please remove [this line](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/09887a13d938c6478babe853c4c88ed4ebccff0a/src/%2Benergy/calculateBoundaryConditions.m#L21-L23)? I moved it here.

Done